shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/watchers/ServiceEvent.java

```diff
@@ -0,0 +1,50 @@
+package com.redhat.service.bridge.shard.operator.watchers;
+
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.client.Watcher;
+import io.javaoperatorsdk.operator.processing.event.DefaultEvent;
+
+public class ServiceEvent extends DefaultEvent {
+
+    private final Watcher.Action action;
+    private final Service service;
+
+    public ServiceEvent(Watcher.Action action, Service service, ServiceEventSource serviceEventSource) {
+        // TODO: this mapping is really critical and should be made more explicit by the java operator sdk
```

Comment:
Given this comment about how critical is, do we have a test to make sure this is always done?

Reply:
This is already covered by our tests: we do have tests that when the BridgeController deploys resources (Deployments, Services) and such resources get modified, the reconcile loop of the controller gets called -> meaning that those resources are linked properly. i.e. we do have integration tests that cover this scenario, but not unit tests.